src/components/structures/RoomView.tsx

```diff
@@ -1208,12 +1211,30 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
             case Action.EditEvent: {
                 // Quit early if we're trying to edit events in wrong rendering context
                 if (payload.timelineRenderingType !== this.state.timelineRenderingType) return;
+                if (payload.event && payload.event.getRoomId() !== this.state.roomId) {
+                    // If the event is in a different room (e.g. all rooms search), we need to view that room first
+                    dis.dispatch<ViewRoomPayload>({
+                        action: Action.ViewRoom,
+                        room_id: payload.event?.getRoomId(),
```

Comment:
We already know that `payload.event` is non-nullish

```suggestion
                        room_id: payload.event!.getRoomId(),
```